### PR TITLE
Removes special characters from temporary filenames

### PR DESCRIPTION
**Pull request by abidlabs**
Removes special characters from temporary filenames so that the files can be served by components. 

Closes: #2468

### CODE REVIEW

File: gradio/processing_utils.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 import subprocess
 import tempfile
 import warnings
+from ctypes import util
```
Comment (freddyaboulton) [inline gradio/processing_utils.py:12]: This is the wrong import

File: gradio/processing_utils.py [hunk 2]
```diff
@@ -329,6 +330,7 @@ def create_tmp_copy_of_file(file_path, dir=None):
     if "." in file_name:
         prefix = file_name[0 : file_name.index(".")]
         extension = file_name[file_name.index(".") + 1 :]
+    prefix = utils.strip_invalid_filename_characters(prefix)
```
Comment (freddyaboulton) [inline gradio/processing_utils.py:333]: Would be good to add a unit test for this!

**Review by freddyaboulton — APPROVED** [review]
Looks good to me @abidlabs ! Thanks for the fix.

**Comment by abidlabs** [comment]
Thanks for the review @freddyaboulton! Used this excuse to organize the `test\test_processing_utils.py` a bit

**Comment by ColtonBehannon** [comment]
I am curious if there is any way to still get the special characters out of the file names with fileobj.name? If I want my application to be able to upload that same file to other resources such Azure Storage Explorer I need to be able to access the original file name but the current workflow appears to remove special characters such as the "()" in test_file(1).txt.

**Comment by abidlabs** [comment]
Not possible at the moment @ColtonBehannon since these are stripped. Is it possible for you to strip the same special characters on your end?

**Comment by ColtonBehannon** [comment]
@abidlabs Thanks for the quick response! I think that could work though I'm not sure what all characters are considered special in this context and would need to be stripped. For example, are underscores included as well or not. Any suggestions here other than testing them all?

**Comment by abidlabs** [comment]
Here's the logic: 

https://github.com/gradio-app/gradio/blob/52f7831751b432411e109bd41add4ab286023a8e/client/python/gradio_client/utils.py#L421

so yes underscores are valid

**Pull request merged**